code/game/objects/structures/wallmounts/wall_cabinet.dm

```diff
@@ -0,0 +1,145 @@
+/obj/structure/wall_cabinet
+	name = "wall cabinet"
+	desc = "A small wall mounted cabinet designed to hold small items."
+	icon = 'icons/obj/wallmounts.dmi'
+	icon_state = "cabinet_closed"
+	anchored = 1
+	density = 0
+	var/opened = 0										// Is the cabinet opened? 1/0
+	var/nick = "cabinet"								// Used in sprite-calling; Name this the same as the first part of the icon_states
+	var/capacity = 1									// How many items fit inside? Use a fitting integer
+	var/exclusivity	= 0									// Are multiple items of the same type allowed? 1/0
+	var/allowedContent[] = list(/obj/item)				// What items are allowed inside? Put an item path here, i.e. /obj/item/*
+	var/spawnContent[] = newlist()						// What items should a spawned cabinet start with? Put item paths here, i.e. /obj/item/*
+
+
+/obj/structure/wall_cabinet/New(loc, ndir, building)	// Initializes a new wall cabinet
+	..()
+	if(building)
+		setDir(ndir)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -27 : 27)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
+		opened = 1
+		contents = null
+		update_icon()
+	else
+		for(var/i=1, i <= spawnContent.len, i++)		// If any spawn items are set, run this loop
+			contents += spawnContent[i]
+
+
+/obj/structure/wall_cabinet/proc/allowedItem(var/x)		// Checks whether the items are allowed for this cabinet type
+	for(var/i=1, i <= allowedContent.len, i++)			// If only specific contents are allowed, run this loop
+		if(istype(x, allowedContent[i]))				// Checks if the item fits in the set categories
+			return 1
+	return 0
```

Comment:
`if(var/obj/item/held_item.type in allowedContent)` This whole proc is unnecessary